enhancements/insights/insights-config-api.md

```diff
@@ -0,0 +1,315 @@
+---
+title: insights-config-api
+authors:
+  - "@tremes"
+reviewers:
+  - "@bparees"
+  - "@deads2k"
+  - "@mfojtik"
+approvers:
+  - "@bparees"
+  - "@deads2k"
+  - "@mfojtik"
+api-approvers:
+  - "@deads2k"
+  - "@mfojtik"
+creation-date: 2022-02-16
+last-updated: 2022-04-22
+status: implementable
+tracking-link: 
+  - "https://issues.redhat.com/browse/CCXDEV-6852"
+see-also:
+  - "https://issues.redhat.com/browse/CCXDEV-6852"
+  - "https://issues.redhat.com/browse/CCX-195"
+---
+
+# Define and expose Insights configuration API
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [x] User-facing documentation is created/updated in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement introduces new Insights API resources. The new API will introduce new configuration options (including an option to force the data gathering as well as to disable it completely) as well as status reporting of the Insights operator. 
+
+## Motivation
+
+The main motivation for this new API is to allow users to refresh (or force) the gathering of the Insights data (see the [User stories](#user-stories) section for more details) as well as to disable the gathering completely. This new API will allows us to report various Insights status reports (including the Insights analysis report that is now exposed via Prometheus metrics).
+
+### Goals
+
+The goal of this enhancement is to propose a basic structure for a new Insights configuration API and define new configuration options. This includes following:
+
+- Define a new configuration option providing a way to trigger on demand Insights gathering
+- Define a new configuration option providing a way to include/exclude particular Insights data gatherers
+- Provide Insights status & reporting in the status field in the new API
+
+### Non-Goals
+
+- Move the majority of the current configuration options to a new configuration API - Insights has quite a lot of configuration options
+  and it does not make sense to expose everything in the new configuration API 
+- Removing the original configuration via `support` secret
```

Comment:
Any particular reason why not? What about the precedence? Ie. what is more important the `support` secret or the API configuration if they are touching similar areas?

Reply:
This is described in the https://github.com/openshift/enhancements/pull/1037/files#diff-e44167fc640994cf3d2a0afef771e4f11d3be51ccf7e5209c1cd8c9c81e054a1R260 and in the https://github.com/openshift/enhancements/pull/1037/files#diff-e44167fc640994cf3d2a0afef771e4f11d3be51ccf7e5209c1cd8c9c81e054a1R268

Reply:
I'd like this `"support" secret config takes precedence over the new config API` to be explicitly mentioned right below after you introduce this new API. Test plan is fine, but that's not where I look for key assumptions.

Reply:
I added it to https://github.com/openshift/enhancements/pull/1037/files#diff-e44167fc640994cf3d2a0afef771e4f11d3be51ccf7e5209c1cd8c9c81e054a1R117 ...not sure if it's the best place....

Reply:
I'd suggest moving it at the top of the Proposal section, it's lost between API types.

Reply:
OK. Moved.